From 7eaff0ef8a754969c7423cd488370345521b47a1 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 29 Jul 2025 14:49:41 +0200
Subject: [PATCH] drm/panel: otm8009a: Check  gpio reset

This fix allows you to check the status of the GPIO reset.
This GPIO can be configured by another software layer.
To maintain continuous display, it is best not to reset the panel
if it has already been started.

Change-Id: I4c627b66d03d37007fad3a8b726eaa16d3926e84
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479218
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/panel/panel-orisetech-otm8009a.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
index ad88517198e7..157c3e05e17b 100644
--- a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
+++ b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
@@ -454,11 +454,17 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 	if (!ctx)
 		return -ENOMEM;
 
-	ctx->reset_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_LOW);
+	ctx->reset_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_ASIS);
 	if (IS_ERR(ctx->reset_gpio))
 		return dev_err_probe(dev, PTR_ERR(ctx->reset_gpio),
 				     "cannot get reset GPIO\n");
 
+	if (ctx->reset_gpio) {
+		ret = gpiod_get_direction(ctx->reset_gpio);
+		if (ret != 0)
+			gpiod_direction_output(ctx->reset_gpio, 1);
+	}
+
 	ctx->supply = devm_regulator_get(dev, "power");
 	if (IS_ERR(ctx->supply))
 		return dev_err_probe(dev, PTR_ERR(ctx->supply),
-- 
2.53.0

